### PR TITLE
RavenDB-12396: Report will now include a histogram of tree balance depth

### PR DESCRIPTION
**Pull request by redknightlois**

### CODE REVIEW

File: src/Voron/Debugging/StorageReportGenerator.cs [hunk 1]
```diff
@@ -210,19 +211,20 @@ public static TreeReport GetReport(FixedSizeTree fst, bool calculateExactSizes)
...
+        public static TreeReport GetReport(Tree tree, bool includeDetails)
         {
             List<double> pageDensities = null;
-
-            if (calculateExactSizes)
+            Dictionary<int, int> pageBalance = null;
+            if (includeDetails)
             {
                 pageDensities = GetPageDensities(tree);
+                pageBalance = GatherBalanceDistribution(tree);
```
Comment (arekpalinski) [inline src/Voron/Debugging/StorageReportGenerator.cs:227]: I wonder if we could calculate both page densities and balance distribution at once and this way iterate over all pages of a tree one time

Comment (redknightlois) [reply src/Voron/Debugging/StorageReportGenerator.cs:227]: No difference, the execution time is dominated hard by hard page faults. On codealike database it takes a minute the first run. Second is under a second. Also I was under the impression that the details enabled is not usually call.

**Review by arekpalinski — APPROVED** [review]

**Review by ayende — APPROVED** [review]

**Comment by redknightlois** [comment]
Side question, I could optimize the process because I understand the underlying behavior. Would it make sense to do so as an additional ticket?

**Comment by ayende** [comment]
What do you mean? How often do you think we'll be calling something like this?

**Comment by redknightlois** [comment]
Not at all, I use it for debugging... but I dont know if it has any other use.

**Comment by ayende** [comment]
no point in optimizing it if it is fast enough that on your scenario it takes only a minute

**Pull request merged**